psd-web/app/models/investigation.rb

```diff
@@ -58,8 +58,8 @@ class Investigation < ApplicationRecord
   has_one :source, as: :sourceable, dependent: :destroy
   has_one :complainant, dependent: :destroy
 
-  has_many :collaborators, dependent: :destroy
-  has_many :teams, through: :collaborators
+  has_many :edit_accesses, dependent: :destroy, class_name: "Collaboration::EditAccess"
+  has_many :editors, through: :edit_accesses, dependent: :destroy, source: :editor, source_type: "Team"
```

Comment:
Think this should reference "teams" still, to make it clear that it always returns an array of teams, not users (which you might expect from the term "editors").

```suggestion
  has_many :teams_with_edit_access, through: :edit_accesses, dependent: :destroy, source: :editor, source_type: "Team"
```